core/templates/dev/head/pages/exploration-editor-page/services/exploration-rights-data.service.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A data service that stores data
+ * about the rights for this exploration.
+ * This is different from the exploration rights service
+ * since the data used here is required globally and may
+ * create a circular dependency with exploration data service
+ * which is used in exploration rights service
+ */
+require('services/ContextService.ts');
+require('services/contextual/UrlService.ts');
+var oppia = require('AppInit.ts').module;
+
+oppia.factory('ExplorationRightsDataService', [
+  '$http', '$q', 'ContextService', 'UrlService',
+  function($http, $q, ContextService, UrlService) {
+    var rights = null;
+    var pathname = UrlService.getPathname();
+
+    return {
+      getRightsAsync: function() {
+        if (pathname.includes('collection_editor')) {
+          return $q.resolve({});
+        }
```

Comment:
Could you explain this change?

Reply:
I noticed that the collection editor calls some exploration editor services and one of those services depends on the rights data. Leaving it the other way will mean sending a request with collection id instead of exploration id and will return 404.
So I did this to prevent that from happening.

Reply:
Done. #7221 explains more about the issue

Reply:
Can you please reply to my comment in #7221?

Reply:
Okay. I didn't see that earlier.